### PR TITLE
fixing the way we generate the dev docker to run the pylint and pytest

### PR DESCRIPTION
**Pull request by wissamg**
## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Content PR
https://github.com/demisto/content/pull/17746

## Description
After moving to debian bullseye installing pylint and pytest requires gcc to be installed

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/lint/templates/dockerfile.jinja2 [hunk 1]
```diff
@@ -13,7 +13,9 @@ RUN chmod -R 775 /devwork
 {# Install requirments and missing deps #}
 RUN printf "{{ pypi_packs | join('\\n') }}" > /devwork/test-requirements.txt
 RUN OS_RELEASE=$(cat /etc/os-release); if echo "$OS_RELEASE" | grep -q "alpine"; \
-then apk add --no-cache --virtual .build-deps python3-dev gcc build-base; fi; \
+then apk add --no-cache --virtual .build-deps python3-dev gcc build-base; \
+elif echo "$OS_RELEASE" | grep -qi "Debian"; \
+then apt-get update && apt-get install -y --no-install-recommends gcc python3-dev;fi; \
 pip install --no-cache-dir -r /devwork/test-requirements.txt; \
 if echo "$OS_RELEASE" | grep -q "alpine"; then apk del .build-deps; fi;
```
Comment (glicht) [inline demisto_sdk/commands/lint/templates/dockerfile.jinja2:20]: We should uninstall gcc, python3-dev in Debian as we do with apline

**Review by glicht — CHANGES_REQUESTED** [review]

File: demisto_sdk/commands/lint/templates/dockerfile.jinja2 (hunk 1 above)
Comment (wissamg) [reply demisto_sdk/commands/lint/templates/dockerfile.jinja2:20]: > We should uninstall gcc, python3-dev in Debian as we do with apline

yes you are right

File: demisto_sdk/commands/lint/templates/dockerfile.jinja2 [hunk 2]
```diff
@@ -13,9 +13,13 @@ RUN chmod -R 775 /devwork
...
 RUN OS_RELEASE=$(cat /etc/os-release); if echo "$OS_RELEASE" | grep -q "alpine"; \
-then apk add --no-cache --virtual .build-deps python3-dev gcc build-base; fi; \
+then apk add --no-cache --virtual .build-deps python3-dev gcc build-base; \
+elif echo "$OS_RELEASE" | grep -qi "Debian"; \
+then apt-get update && apt-get install -y --no-install-recommends gcc python3-dev;fi; \
 pip install --no-cache-dir -r /devwork/test-requirements.txt; \
-if echo "$OS_RELEASE" | grep -q "alpine"; then apk del .build-deps; fi;
+if echo "$OS_RELEASE" | grep -q "alpine"; then apk del .build-deps; \
+elif echo "$OS_RELEASE" | grep -qi "Debian"; \
+then apt-get autoremove -y gcc python3-dev;fi;
```
Comment (glicht) [inline demisto_sdk/commands/lint/templates/dockerfile.jinja2:22]:
```suggestion
then apt-get purge -y --auto-remove gcc python3-dev; fi;
```

you should use `purge` from what I know

**Review by glicht — CHANGES_REQUESTED** [review]
Please make sure to test before doing a release

**Review by jochman — APPROVED** [review]

**Review by glicht — APPROVED** [review]

**Pull request merged**